.github/workflows/ci.yml

```diff
@@ -87,3 +87,8 @@ jobs:
         if [[ "$ERROR_COUNT" -gt "0" ]]; then
           exit 1
         fi
+    - name: Test Custom Cloud Shell Image Build
+      run: |
+        set -x
+        # build cloud shell image: it will automatically exit if there's an error in build
+        docker build -t custom-cloud-shell:$GITHUB_SHA ./cloud-shell
```

Comment:
Nit: It seems a bit misleading to label the container with `:$GITHUB_SHA`. That would make me think the container is being saved/used somewhere, but the real purpose of this step is the build process, not the artifact itself. 

Maybe use `cloud-shell`, or even `_`?